exceptions/exception_examples.py

```diff
@@ -0,0 +1,108 @@
+"""Handling exceptions
+   - catch multiples exceptions
+   - catch exceptions with arguments
+   - re-raising exception to get stack"""
+
+
+def catch_my_exceptions():
+    try:
+        pass
+    except (ZeroDivisionError, TypeError, RuntimeError):
+        print('Exception found.')
+
+
+def catch_exceptions_with_arguments():
+    try:
+        raise Exception('first argument', 'second argument')
+    except Exception as exc_instance:
+        print(type(exc_instance))  # class type
+        print(exc_instance.args)
+        first, second = exc_instance.args  # accessible arguments
+        print('First: ' + first)
+
+
+def reraise():
+    import sys
+
+    try:
+        f = open('myfile.txt')
+        s = f.readline()
+        i = int(s.strip())
+    # except OSError as err:
+    #     print("OS error:", err)
+    # except ValueError:
+    #     print("Could not convert data to an integer.")
+    except Exception as err:  # catch any exception and print exception class name
+        print(f"Unexpected {err=}, {type(err)=}")
+        raise  # re-raise the exception for caller, it will show stack trace
+
+
+"""Raising exceptions -  forcing exception to raise"""
+
+
+def raise_specified_exception():
+    try:
+        raise TimeoutError('Timeout!')
+    except TimeoutError:
+        print('Timeout exception!')
+        raise
+
+
+"""Chaining exceptions
+   + cleanup with finally"""
+
+
+def chain_one_exception_to_another():
+    try:
+        return 1/0
+    except ZeroDivisionError:  # catch zero division error
+        raise RuntimeError('cannot run it under such conditions')  #raise runtime error
+    finally:  # clean-up: print and raise another error
+        print('Cleanup action')
+        raise KeyboardInterrupt
```

Comment:
what's the intention of raising the exception here?

Reply:
I think I got carried away with raising more exceptions. For one, it's not doing anything here logically. Also, I've chosen fairly random exception, as KeyboardInterrupt makes sense to use when e.g. I'm in a infinite loop and I want to handle user interrupting.